### PR TITLE
Sort out generated features for landuse, buildings & POIs

### PR DESCRIPTION
**Pull request by zerebubuth**
* Added town hall POIs.
* Updated building selection criteria - turns out there's a lot of `building=no` things.
* Separated buildings, landuse and POIs so that the name should come through in only one of these layers. The polygon geometry is only duplicated between landuse and buildings when generating a landuse label, after which the landuse polygon is discarded. This gets rid of many duplicate labels.
* Added config to generate address points for (and from) buildings and drop landuse polygons which are duplicating buildings.

Connects to #201.

@rmarianski could you review, please?

### CODE REVIEW

**Comment by zerebubuth** [comment]
I should have said: Requires mapzen/TileStache#72 also.

File: queries/buildings.jinja2 [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
+    -- building.
+    CASE WHEN mz_poi_min_zoom IS NULL AND
+              (mz_is_landuse IS NULL OR mz_is_landuse = FALSE)
+         THEN name
+         ELSE "addr:housename"
+    END AS name,
     way_area::bigint AS area,
     osm_id AS __id__,
-    "building:part",
+    'openstreetmap.org' AS source,
```
Comment (nvkelso) [inline queries/buildings.jinja2]: +1

File: data/functions.sql [hunk 2]
```diff
@@ -161,6 +161,7 @@ BEGIN
       WHEN historic_val = 'landmark'         THEN LEAST(zoom + 1.76, 15)
       WHEN leisure_val  = 'marina'           THEN LEAST(zoom + 3.45, 17)
       WHEN amenity_val  = 'place_of_worship' THEN LEAST(2 * zoom - 9.55, 17)
+      WHEN amenity_val  = 'townhall'         THEN LEAST(zoom + 1.85, 16)
```
Comment (nvkelso) [inline data/functions.sql]: Good add!

File: data/functions.sql [hunk 3]
```diff
@@ -269,7 +270,11 @@ CREATE OR REPLACE FUNCTION mz_calculate_is_building_or_part(
     building_val text, buildingpart_val text)
 RETURNS BOOLEAN AS $$
 BEGIN
-    RETURN (building_val IS NOT NULL OR buildingpart_val IS NOT NULL);
+    -- there are 12,000 uses of building=no, so we ought to take that into
+    -- account when figuring out if something is a building or not. also,
+    -- returning "kind=no" is a bit weird.
+    RETURN ((building_val IS NOT NULL AND building_val <> 'no') OR
+            (buildingpart_val IS NOT NULL AND buildingpart_val <> 'no'));
```
Comment (rmarianski) [inline data/functions.sql]: Nice catch!

File: queries/buildings.jinja2 (hunk 1 above)
Comment (rmarianski) [inline queries/buildings.jinja2]: For buildings the source will always be 'openstreetmap.org`, no? Is this for consistency with other layers? Or would a post process step create features that have a different source or none at all?

Comment (zerebubuth) [inline queries/buildings.jinja2]: I don't know. I _thought_ I was adding it for compatibility with the other layers, but it seems that we don't include `source` for the boundaries, earth, POIs or transit layers either. Transit and POIs are currently OSM-only, but earth and boundaries do mix data sources...

@nvkelso Should we be annotating the source for _all_ features, or just features in layers which aren't single-source?

Comment (nvkelso) [inline queries/buildings.jinja2]: In the not too distant future we may also offer Polygon City building data.
So we can either start doing this now, or then.

**Comment by rmarianski** [comment]
:+1:

**Pull request merged**